### PR TITLE
Fix bug for not able to get sourceTables from metadata

### PR DESCRIPTION
**Pull request by seankao-az**
### Description
Using custom implementation for `FlintIndexMetadataService` exposes a bug where we incorrectly get empty `sourceTables` from `FlintMetadata`.
Depending on how the `FlintMetadata` is generated, the `metadata.properties.get("sourceTables")` could either be of type java array or scala array.
This PR fixes the bug, now considering both cases.

Also added some logs for troubleshooting in production environment.

### Related Issues
* #746 
* #854 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/sql/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: flint-spark-integration/src/main/scala/org/opensearch/flint/spark/mv/FlintSparkMaterializedView.scala [hunk 1]
```diff
@@ -179,13 +181,43 @@ object FlintSparkMaterializedView {
...
+   * Get source tables from Flint metadata properties field.
+   *
+   * @param metadata
+   *   Flint metadata
+   * @return
+   *   source table names
+   */
+  def getSourceTablesFromMetadata(metadata: FlintMetadata): Array[String] = {
+    logInfo(s"Getting source tables from metadata $metadata")
+    metadata.properties.get("sourceTables") match {
```
Comment (ykmr1224) [inline flint-spark-integration/src/main/scala/org/opensearch/flint/spark/mv/FlintSparkMaterializedView.scala:206]: Can `metadata.properties` to be null?

**Review by dai-chen — COMMENTED** [review]
Is it possible to enforce a single and language-independent format in metadata, e.g. JSON array?

File: flint-spark-integration/src/main/scala/org/opensearch/flint/spark/mv/FlintSparkMaterializedView.scala (hunk 1 above)
Comment (seankao-az) [reply flint-spark-integration/src/main/scala/org/opensearch/flint/spark/mv/FlintSparkMaterializedView.scala:206]: When FlintMetadata is built before passing in, we add a `properties` field even if it's empty..
We could still do a null check just to be safe.

**Comment by seankao-az** [comment]
> Is it possible to enforce a single and language-independent format in metadata, e.g. JSON array?

Not quite sure..
Since when reading from OpenSearch index mappings, we use jackson library to deserialize into `FlintMetadata`, and jackson library will use `java.util.ArrayList`. I'll try using this type instead.

Or do you mean using json string? Like
```
{"sourceTables": "[\"a.b.c\"], [\"d.e.f\"]"}
```

**Comment by dai-chen** [comment]
> > Is it possible to enforce a single and language-independent format in metadata, e.g. JSON array?
> 
> Not quite sure.. Since when reading from OpenSearch index mappings, we use jackson library to deserialize into `FlintMetadata`, and jackson library will use `java.util.ArrayList`. I'll try using this type instead.
> 
> Or do you mean using json string? Like
> 
> ```
> {"sourceTables": "[\"a.b.c\"], [\"d.e.f\"]"}
> ```

Yeah, just trying to understand where this problem comes from? Is Scala or Java array stored in metadata?

**Comment by seankao-az** [comment]
mainly comes from this line of scala code. I was putting scala Array[String] here.
https://github.com/opensearch-project/opensearch-spark/pull/883/files#diff-1e6cbb4cde65f1142862f3b539a438d534fd70a229499efd671ef9aae67ff29fR75

Now I've changed it to use ArrayList. so for both cases (`mv.metadata()` and deserialized metadata from index mapping) will have ArrayList. Do you think this is better? @dai-chen

**Review by dai-chen — APPROVED** [review]

**Pull request merged**